R/usage_to_annual_nets_distributed.R

```diff
@@ -0,0 +1,234 @@
+    stop("Target usage(s) cannot be achieved with estimated use rate in any of the countries. Refer to maximum usage above.")
+  } else if(any(target_usage>min(country_use_rate[country_use_rate$iso3 %in% country_iso3,"use_rate"]))) {
+    print(max_usage)
+    print("Target usage(s) cannot be achieved with estimated use rate in some of the countries - return NA. Refer to maximum usage above.")
+  }
+  
+  targets_all <- lapply(target_usage, function(this_use) {
+    targets_dt <- country_use_rate[country_use_rate$iso3 %in% country_iso3, 
+                                   c("iso3", "year", "use_rate")]
+    targets_dt$target_use <- this_use
+    # Convert target usage to target access: access = usage/use rate
+    # since use rate = the proportion of people with access to a net who slept under it
+    targets_dt$target_access <- targets_dt$target_use/targets_dt$use_rate
+    targets_dt <- targets_dt[order(targets_dt$target_access),]
+    targets_dt$target_percapita_nets <- NA
+    targets_dt$target_percapita_nets[
+      targets_dt$target_access<=max(loess_for_prediction$loess_predicted_access)] <- 
+      approx(loess_for_prediction$loess_predicted_access,
+             loess_for_prediction$percapita_nets, 
+             targets_dt$target_access[targets_dt$target_access<=max(loess_for_prediction$loess_predicted_access)])$y
+    # Relationship between access and NPC is aggregated for all countries,
+    # but use rate to convert usage to access is country-specific (derived from spatiotemporal regression?)
+    return(targets_dt[order(targets_dt$iso3),])
+  })
+  targets_all <- do.call("rbind", targets_all)
+  targets_all$target_access[targets_all$target_access>1] <- NA
+  # Set access to NA where it is over 1 and excluded from conversion
+  
+  ### 2. Convert equilibrium nets per capita to annual nets distributed per capita ###
+  
+  # Read in net half lives (median retention times) for each country
+  half_life_data <- data$half_life_data
+  
+  # Estimate time at which all nets are lost (parameter l) based on net half life in each country
+  half_life_data$l <- half_life_data$half_life/sqrt(1 - k / (k - log(0.5)))
+  
+  # Merge this with target nets per capita by country and target usage
+  nets_distributed <- merge(x=targets_all, y=half_life_data[,c("iso3", "l")], 
+                            by = "iso3", all.x=TRUE, all.y=FALSE)
+  
+  # Use net loss function to calculate annual nets distributed per capita for different countries 
+  # and target usages
+  # The annual number of nets distributed to maintain the input equilibrium nets per capita. 
+  # This is for the specific "smoothed" case where we assume that 1/3 of the population at risk
+  # is distributed nets every 3 years (this is equivalent to a smoothed 3 yearly whole 
+  # population distribution).
+  nets_distributed$annual_percapita_nets_distributed <- nets_distributed$target_percapita_nets * 
+    integrate_net_loss_vector(k=k, l = nets_distributed$l) * (1/3)
```

Comment:
This conversion is assuming equilibrium distributions of equial size on a regular schedule.

Eventually I think we might want to "fit" the annual_percapita_nets_distributed to match the target_per_capita nets over time. This is an (almost) identical operation to the fitting the malariasimulation input distirbutions to match target population usage over time - so we should be able to use the same functionality.

Reply:
I will add this as an issue